### PR TITLE
Base image build pipeline - Initial commit

### PR DESCRIPTION
**Pull request by rootAvish**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM centos:7
+MAINTAINER Avishkar Gupta <avgupta@redhat.com>
```
Comment (jpopelka) [inline Dockerfile]: https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+RUN yum install -y epel-release && \
+    yum install -y zip && \
+    yum install -y python-pip python-devel gcc && \
+    yum -y install tkinter && \
+    yum-config-manager --disable testing-devtools-2-centos-7 && \
+    yum -y install gcc-c++.x86_64 && \
```
Comment (jpopelka) [inline Dockerfile]: Can we install all the packages with one `yum` call ?
```
RUN yum -y install -y epel-release && \
    yum-config-manager --disable testing-devtools-2-centos-7 && \
    yum install -y zip python-pip python-devel gcc gcc-c++.x86_64 tkinter && \
    yum clean all
```

Why do we disable the `testing-devtools-2-centos-7` repo ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    yum clean all
+
+
+# --------------------------------------------------------------------------------------------------
+# install python packages
+# --------------------------------------------------------------------------------------------------
+COPY ./hack/requirements.txt /
+RUN pip install -r /requirements.txt && rm /requirements.txt
+RUN pip install pomegranate==0.7.3
```
Comment (jpopelka) [inline Dockerfile]: Why is this here and not in `requirements.txt` ?

**Review by jpopelka — COMMENTED** [review]
I don't think we need Jenkinsfile at the moment. Current CI/CD does not use it.

File: Dockerfile (hunk 3 above)
Comment (rootAvish) [reply Dockerfile]: There's an upstream bug with dependency resolution for this package, we tried doing it in Kronos: https://github.com/fabric8-analytics/fabric8-analytics-stack-analysis/pull/68

Comment (jpopelka) [reply Dockerfile]: OK, then put a comment there so other people don't have to ask the same question.

File: Dockerfile (hunk 2 above)
Comment (rootAvish) [reply Dockerfile]: I took this from the Kronos Dockerfile we had, assuming it had a reason but didn't read, apologies for that. From what it looks like the reason might have something to do with installation of gcc_c++ like here: https://stackoverflow.com/questions/24634776/installing-multiple-gcc-on-centos-7. I tried a build with that line removed and it passed, so I'm assuming whatever issue was there has been fixed.

I'll remove this as well.

**Comment by rootAvish** [comment]
@jpopelka thanks for the review, I guess everything should look okay now.

**Review by jpopelka — APPROVED** [review]
Let me know once https://github.com/almighty/almighty-jobs/pull/413 is merged (or feel free to merge yourself).

**Comment by rootAvish** [comment]
[test]

**Comment by rootAvish** [comment]
[test this]

**Comment by rootAvish** [comment]
I think I've missed a step somewhere, can't seem to get it to build.

**Comment by rootAvish** [comment]
[test]

**Comment by rootAvish** [comment]
[test]

**Comment by rootAvish** [comment]
[test]

**Comment by rootAvish** [comment]
Build is up on Centos-CI even though we can't see status. I guess merge build too will succeed.
